### ISSUE TITLE
Deriver: Boxable

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#2159]
Automatically derive an implementation of `Boxable.box` for each enum marked with Boxable.

*** Moreover, if an enum is marked with `Eq`, `Order`, and `ToString` then automatically mark it with Boxable. ***

**Comment by mlutze** [comment]
> Moreover, if an enum is marked with Eq, Order, and ToString then automatically mark it with Boxable.

This feels a little "magicky" to me. What's the reasoning for it?

**Comment by magnus-madsen** [comment]
> > Moreover, if an enum is marked with Eq, Order, and ToString then automatically mark it with Boxable.
> 
> This feels a little "magicky" to me. What's the reasoning for it?

![image](https://user-images.githubusercontent.com/2287953/130787853-50b773d2-c021-4e1e-9778-73ec8a070377.png)

**Comment by magnus-madsen** [comment]
TLDR: Flix hides the fact that there are primitive and non-primitive values. In Flix there is no difference.

Except for Boxable. Hence we will hide it too!

**Comment by mlutze** [comment]
Should `Boxable` be implicit if there are non-derived instances of `ToString`, `Eq`, and `Order`?

**Comment by magnus-madsen** [comment]
> 
> 
> Should `Boxable` also be implicit if there are non-derived instances of `ToString`, `Eq`, and `Order`?

What do you mean?

**Comment by mlutze** [comment]
I understand that the following should be equivalent:
```scala
enum E with Eq, Order, ToString
// vs
enum E with Eq, Order, ToString, Boxable
```

Should these also be equivalent?
```scala
enum E
instance Eq[E] { /* ... */ }
instance Order[E] { /* ... */ }
instance ToString[E] { /* ... */ }
// vs
enum E with Boxable
instance Eq[E] { /* ... */ }
instance Order[E] { /* ... */ }
instance ToString[E] { /* ... /* }
```

**Comment by magnus-madsen** [comment]
Hmm. Would it be difficult to make them equivalent? If not, I think we should go for it. To hide Boxable as much as possible!

**Comment by mlutze** [comment]
Actually it might not be so straightforward. It would have to be done after getting proper types, outside the dériver. So I guess we can consider it future work.
